### ISSUE TITLE
Are there upstream interests in continuing to maintain this repository or is it feature-complete ? 

### ISSUE THREAD

**Issue by TilakMaddy** [ebkalderon/tower-lsp#427]

**Comment by idealseal** [comment]
I don't think this is able to be feature complete any time soon, aside from bumping dependencies, unless the [LSP specification](https://microsoft.github.io/language-server-protocol/) is going to be. I have interest in keeping it alive,  and this is already a couple of versions behind [`lsp-types`](https://github.com/gluon-lang/lsp-types), which implements the basic building blocks of a LSP. I'll fork it for now, if this repository is getting active again I will try to merge my commits here.

**Comment by trevor-scheer** [comment]
@idealseal do you intend to publish your fork? We're actively using this crate but can't take a git dependency unfortunately.

**Comment by TilakMaddy** [comment]
@trevor-scheer `cargo install --git <url>` works no ?

**Comment by idealseal** [comment]
@TilakMaddy I think it's rather a policy on their side. In any other case, `cargo install/add --git` should work fine.

**Comment by idealseal** [comment]
@trevor-scheer I thought I will give @ebkalderon a little time, but I can publish the fork if there is interest already.

**Comment by trevor-scheer** [comment]
@idealseal thanks, we can wait a bit 👍

@TilakMaddy we publish to crates.io so can't have git dependencies.

https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-other-registries

**Comment by idealseal** [comment]
Any ideas for a library name? Otherwise I would go for something like `tower-lsp2`, but I'm not sure about that.

**Comment by trevor-scheer** [comment]
@idealseal that seems fine. I can't think of anything better that isn't a joke (`2tower-2lsp` 🚘🚔).

I've sent Eyal an email, perhaps we wait til next week? No urgency on my end. I'm hoping that if he's not interested in maintaining that maybe he would enable others to 🤞

**Comment by idealseal** [comment]
We __need__ something funny. Otherwise this wouldn't make sense to maintain... :crab:

**Comment by idealseal** [comment]
That's an amazing name, I need more of that sort. I will think about it :laughing:

**Comment by sh-cho** [comment]
`hightower-lsp` maybe.. 😄

**Comment by trevor-scheer** [comment]
I'm off to my wedding and honeymoon for ~ a month so won't be too responsive if at all. I haven't received a response to my email yet unfortunately.

There is still no urgency for this on my end though. My colleague @Jephuff will follow along for updates. Just as a consideration for your own efforts, if it's got to be forked, it's possible we'll need to maintain our own in the long term.

**Comment by TilakMaddy** [comment]
We're using this crate as well for https://github.com/Cyfrin/aderyn would love to chime in and support when I can !

**Comment by ratmice** [comment]
> I don't think this is able to be feature complete any time soon, aside from bumping dependencies, unless the [LSP specification](https://microsoft.github.io/language-server-protocol/) is going to be. I have interest in keeping it alive, and this is already a couple of versions behind [`lsp-types`](https://github.com/gluon-lang/lsp-types), which implements the basic building blocks of a LSP. I'll fork it for now, if this repository is getting active again I will try to merge my commits here.

I feel I must say that this seems backwards from the way things are normally done, make a bunch of PRs and then if those don't get merged, consider forking.

**Comment by trevor-scheer** [comment]
@ratmice I think a good faith effort is being made here. This issue is 2mos old with no response from the maintainer. We're making legitimate efforts to avoid forking if possible and I've reached out to him directly. It's true there aren't open PRs for the dependency updates but given the lack of response so far, I wouldn't expect those to be answered either.

**Comment by idealseal** [comment]
> It's true there aren't open PRs for the dependency updates [...]

Well, there are: #401 #412 #426

**Comment by ratmice** [comment]
@trevor-scheer My main concern is that the user proposing forking is not following the typical process of making PRs themself before forking, along with the account proposing to fork only being a month old, newer than this issue itself! 

While everyone must start somewhere, I do find it hard not to be skeptical. LSPs could make for an attractive target for some purposes, and no one is going to make it obvious they are making a bad faith effort.  The above 2 things are enough for me to nope out on this proposed fork well intended as it might be.

**Comment by sh-cho** [comment]
Maybe make an organization for `tower-lsp` and designate some maintainers? I think we all understand concern of forking in many ways but also `tower-lsp`'s lack of maintenance. The show must go on.

There was a similar problem for `exa`. People moved to [eza-community/eza](https://github.com/eza-community/eza)
- https://github.com/ogham/exa/issues/1243

**Comment by trevor-scheer** [comment]
@ratmice ah, I think your concerns about the potential for a bad actor are valid, admittedly that was not where I was coming from when I said "good faith".

I'd be happy to create that GH org and add contributors if that's the route people want to go. I don't know that anything in particular qualifies me over someone else to admin the org/fork but I'd be happy to do it in the absence of anyone else who felt particularly motivated to. My interest in this project lies in the current work I do for Apollo Graph.

An alternative option would be to find a relevant existing org to be the "host" if they're interested, i.e.:
* https://github.com/gluon-lang (owners of `lsp-types`) @Marwes @brendanzab
* https://github.com/tower-rs (owners of `tower`) @carllerche @LucioFranco

**Comment by trevor-scheer** [comment]
> > It's true there aren't open PRs for the dependency updates [...]
> 
> Well, there are: #401 #412 #426

Apologies for the assumption on my part 🤐 thank you for the correction!

**Comment by lordspacehog** [comment]
I would also assist with maintenance and continued development here.

**Comment by mrnossiom** [comment]
I don't think `tower-rs` hosting the fork makes much sense as this is a more specialized project. `gluon-lang` seems good, but maybe they don't want this additional project to maintain.

I also need `tower-lsp` for personal projects and would be happy to admin this but like Trevor, I have no major contributions to this repo to qualify. This project has been developed close to entirely by Eyal

For the name : maybe `tower-lzp`, maybe considered like typo squat, but I like the vibe of it. Also matches the exa → eza renaming.

**Comment by Decodetalkers** [comment]
Is there any conclusions? Has anyone got  access to the author?

**Comment by trevor-scheer** [comment]
@Decodetalkers No response from the author unfortunately. I'll be back from holiday near the end of November. At that point I'd be happy to create the org and fork if nobody else has by then. In any case I'd be happy to be involved as a maintainer or admin.

**Comment by mrnossiom** [comment]
> I see that most of the people are interested in a maintained fork, but it seems that no one wants to be the one creating the org nor has taken the first steps. With this in mind, I've created the org [tower-lsp](https://github.com/tower-lsp) to kickstart the effort, and a few of you have already been invited to join.

See https://github.com/tower-lsp/tower-lsp/issues/1

**Comment by Decodetalkers** [comment]
> > I see that most of the people are interested in a maintained fork, but it seems that no one wants to be the one creating the org nor has taken the first steps. With this in mind, I've created the org [tower-lsp](https://github.com/tower-lsp) to kickstart the effort, and a few of you have already been invited to join.
> 
> See [tower-lsp/tower-lsp#1](https://github.com/tower-lsp/tower-lsp/issues/1)

have the name been decided? I want to start contribute to the fork.

Do you need help now?

**Comment by mrnossiom** [comment]
Not yet, but you can join the relevant discussion https://github.com/tower-lsp/tower-lsp/issues/3. 
There's seems to be a consensus for `tower-lsp2`

**Comment by tv42** [comment]
Greetings from 2026. See also https://github.com/oxalica/async-lsp
